### ISSUE TITLE
''move_legend'' lost the markers for lineplot and hatch for barplot

### ISSUE THREAD

**Issue by Milkigit** [mwaskom/seaborn#2731]
Hi there, 

The move_legend indeed makes the DIY legend much more convenient and saves me a lot of effort, however, I have two small issues that bothered me a lot.

Here is my code.

```
g=sns.catplot(x="X",  y="Y", data=data_df, kind="bar", legend=True, facet_kws={"legend_out": False})
sns.move_legend(g, "center", bbox_to_anchor=(.5, 0.05), ncol=3, title=None, frameon=False, numpoints=2)
g.fig.subplots_adjust(bottom=0.25)
```
![image](https://user-images.githubusercontent.com/5469169/150451570-15df0c9a-4d33-4b53-9f90-af0750d23a70.png)
![image](https://user-images.githubusercontent.com/5469169/150451648-11046367-a349-46c4-b2d1-de4caa9cbfd5.png) 

Would appreciate it if you could give me some hints.

**Comment by mwaskom** [comment]
Please share a reproducible example and provide versions of seaborn, matplotlib, etc. as requested in the contributing guide. Thanks!

**Comment by Milkigit** [comment]
Thanks for the fast response, the versions are seaborn==0.11.2, matplotlib==3.4.3, python==3.9.7.

A reproducible example for the line plot is as follows.

```
        titanic = sns.load_dataset("titanic")
        g=sns.relplot(x="sex", y="survived", hue="class", style="class",
                      data=titanic, kind="line", legend=True,  markers=self.marker,
                      facet_kws={"legend_out": False},
                      col='embark_town', 
                      height=3,
                      aspect=1.5)
        plt.rcParams['figure.autolayout'] = True
        plt.savefig("test_move_legend_line.pdf", format='pdf')    
```
![image](https://user-images.githubusercontent.com/5469169/150542542-a41992c9-dec7-4459-82c9-391e12346db0.png)

```
        titanic = sns.load_dataset("titanic")
        g=sns.relplot(x="sex", y="survived", hue="class", style="class",
                      data=titanic, kind="line", legend=True,  markers=self.marker,
                      facet_kws={"legend_out": False},
                      col='embark_town', 
                      height=3,
                      aspect=1.5)
        # move the legend to the bottom.
        sns.move_legend(g, "center", bbox_to_anchor=(.5, 0.05), ncol=6, title=None, frameon=False, numpoints=2)
        g.fig.subplots_adjust(bottom=0.25)
        plt.rcParams['figure.autolayout'] = True
        plt.savefig("test_move_legend_line.pdf", format='pdf')    
```
![image](https://user-images.githubusercontent.com/5469169/150542653-23202d03-8295-4a36-8316-33d09abb90e4.png)

**Comment by Milkigit** [comment]
I found an extra issue in bar plot, the blank space was not automatically removed.
```
        titanic = sns.load_dataset("titanic")
        g=sns.catplot(x="sex", y="survived", hue="class", 
                      data=titanic, kind="bar", legend=True,
                      facet_kws={"legend_out": False},
                      alpha=0.8,
                      col="embark_town",
                      height=3,
                      aspect=1.5)
        hatches = ['-', '+', 'x', '\\', '*', 'o']*3
        for j in range(3):
            for i, thisbar in enumerate(g.axes[0][j].patches):
                # Set a different hatch for each bar
                thisbar.set_hatch(hatches[i])        
        # sns.move_legend(g, "center", bbox_to_anchor=(.5, 0.05), ncol=4, title=None, frameon=False, numpoints=2)
        # g.fig.subplots_adjust(bottom=0.25)
        plt.rcParams['figure.autolayout'] = True
        plt.savefig("test_move_legend_o.pdf", format='pdf')
```
![image](https://user-images.githubusercontent.com/5469169/150555233-cea27ea6-863f-419a-82ec-ec37abba13ea.png)

```
        titanic = sns.load_dataset("titanic")
        g=sns.catplot(x="sex", y="survived", hue="class", 
                      data=titanic, kind="bar", legend=True,
                      facet_kws={"legend_out": False},
                      alpha=0.8,
                      col="embark_town",
                      height=3,
                      aspect=1.5)
        hatches = ['-', '+', 'x', '\\', '*', 'o']*3
        for j in range(3):
            for i, thisbar in enumerate(g.axes[0][j].patches):
                # Set a different hatch for each bar
                thisbar.set_hatch(hatches[i])        
        sns.move_legend(g, "center", bbox_to_anchor=(.5, 0.05), ncol=4, title=None, frameon=False, numpoints=2)
        g.fig.subplots_adjust(bottom=0.25)
        plt.rcParams['figure.autolayout'] = True
        plt.savefig("test_move_legend_move.pdf", format='pdf')
```
![image](https://user-images.githubusercontent.com/5469169/150555197-578d4fe1-e6cb-431d-b454-a1c700817045.png)

**Comment by mwaskom** [comment]
> A reproducible example for the line plot is as follows.

What is `self`? In any case, after editing your code so that I can run it, I am unable to reproduce the issue as I understand it:

![image](https://user-images.githubusercontent.com/315810/150614960-ccd50c30-6000-4e82-8b32-2b72ffc4b90a.png)

For the bar plot, are you expecting the legend to show the patches you are manually adding to the bars? They are not in the legend to begin with so you'd need to modify the legend artists too.

> I found an extra issue in bar plot, the blank space was not automatically removed.

`catplot` has the `legend_out` parameter as part of its signature and i guess that overrides what appears in `facet_kws`.

**Comment by Milkigit** [comment]
Sorry for the typo, actually the `self.` can be removed. 
I updated the code, however, the markers for the line legend are still missing. Would you please show me your modifications?

**Comment by Milkigit** [comment]
After I modify the legend, I have the markers and hatches in legend now. Thanks a lot for the tips. 
I will close this issue accordingly.

**Comment by mwaskom** [comment]
Would you please show me your modifications?

```python
import seaborn as sns
import matplotlib.pyplot as plt

titanic = sns.load_dataset("titanic")
g=sns.relplot(x="sex", y="survived", hue="class", style="class",
              data=titanic, kind="line", legend=True,  markers=['H', 'P', 'X'],
              facet_kws={"legend_out": False},
              col='embark_town', 
              height=3,
              aspect=1.5)
# move the legend to the bottom.
sns.move_legend(g, "center", bbox_to_anchor=(.5, 0.05), ncol=6, title=None, frameon=False, numpoints=2)
g.fig.subplots_adjust(bottom=0.25)
```
![image](https://user-images.githubusercontent.com/315810/150652919-76ea6318-3f82-4383-88ad-92427587770e.png)

I am using matplotlib 3.5.1, so perhaps there is a bug that they fixed in 3.5.

**Comment by Milkigit** [comment]
Thanks for the update.  Interestingly, directly modifying the legend artists of `relplot` also works for me. 
The workaround for the barplot blank is adjusting the subplots, which also solved my problem. 😂 
```
g.fig.subplots_adjust(bottom=0.25, right=1)
```
